app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
```

Comment:
Actually, the `reload` is needed. The `@last_statuses` memoization doesn't come into play now that we're using `significant_status`. But we do need the `reload` because `statuses` is sorted by `created_at: :desc` but when the new `status` is created it's added at the end of the collection. The `reload` forces the sort to happen moving it to the front.

Maybe there's another way to get the new status added in the right place?